### PR TITLE
Add Port Assignment type

### LINKED ISSUE #16: Add support for Port Assignment
### What problem are you facing?

`crossplane-provider-packet` does not currently have support for assigning ports to virtual networks

### How could the Packet Crossplane Provider help solve your problem?

Add `v1alpha1` support for `Assignment` in `ports.packet.crossplane.io` group.

### PR DESCRIPTION
**Pull request by hasheddan**
### Description of your changes

Adds `v1alpha1` port `Assignment` type.

Fixes #16 

### Checklist

I have:
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Ensured this PR contains a neat, self documenting set of commits.
- [ ] Updated any relevant [documentation], [examples], or [release notes].
- [ ] Updated the [package resources documentation] and [`app.yaml`] to include any new managed resources or classes.

[documentation]: https://github.com/crossplane/crossplane/tree/master/docs
[examples]: https://github.com/crossplane/crossplane/tree/master/cluster/examples
[release notes]: https://github.com/crossplane/crossplane/tree/master/PendingReleaseNotes.md
[`app.yaml`]: https://github.com/packethost/crossplane-provider-packet/blob/master/config/package/manifests/app.yaml
[package resources documentation]: https://github.com/packethost/crossplane-provider-packet/blob/master/config/package/manifests/resources

### CODE REVIEW

File: pkg/controller/ports/assignment/managed.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+	for _, net := range port.AttachedVirtualNetworks {
+		if net.ID == a.Spec.ForProvider.VirtualNetworkID {
+			a.Status.SetConditions(runtimev1alpha1.Available())
+			resource.SetBindable(a)
+			o.ResourceExists = true
+		}
+	}
+
+	meta.SetExternalName(a, port.ID)
```
Comment (hasheddan) [inline pkg/controller/ports/assignment/managed.go:143]: This is admittedly a bit of a weird way to use the `ExternalName` annotation here. It is not used in the `Observe` call at all and the assignment itself is not actually an API object on the packet side. However, once we set it we don't have to make `GetPortByName` calls in `Create` and `Delete` anymore. We could just as easily put this into the `Status` and reference it in that way.

Comment (displague) [reply pkg/controller/ports/assignment/managed.go:143]: It makes sense. Maybe Href would be a better field to use, but it doesn't make much difference (and Href would need more handling to use).

File: apis/ports/v1alpha1/assignment_types.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+	// +optional
+	// +immutable
+	DeviceIDRef *runtimev1alpha1.Reference `json:"deviceIdRef,omitempty"`
+
+	// +optional
+	DeviceIDSelector *runtimev1alpha1.Selector `json:"deviceIdSelector,omitempty"`
+
+	// +immutable
+	Name string `json:"portId"`
```
Comment (displague) [inline apis/ports/v1alpha1/assignment_types.go]: Perhaps this should be PortID?

Comment (hasheddan) [reply apis/ports/v1alpha1/assignment_types.go]: Updated to `name` :+1:

**Review by displague — APPROVED** [review]
Thanks again, @hasheddan - LGTM

**Pull request merged**